### PR TITLE
feat: Python `TGeoDetector.create()` can read config from JSON file

### PR DESCRIPTION
**Pull request by timadye**
* `TGeoDetector.create(jsonFile=tgeo.json)` fills the config from the JSON file. Other settings can still be set from Python.
* Fixes a small issue with the `_detector_create` adapter, which didn't recognise the `jsonFile` setter with no getter.

### CODE REVIEW

File: Examples/Python/python/acts/_adapter.py [hunk 1]
```diff
@@ -84,9 +84,9 @@ def create(*args, mdecorator=None, **kwargs):
             cfg = config_class()
         _kwargs = {}
         for k, v in kwargs.items():
-            if hasattr(cfg, k):
+            try:
                 setattr(cfg, k, v)
-            else:
+            except AttributeError:
```
Comment (paulgessinger) [inline Examples/Python/python/acts/_adapter.py:89]: Is `hasattr(cfg, k)` not covering some failure case or why the `try/catch`?

**Comment by paulgessinger** [comment]
In principle, fair enough. But if you're using the python bindings anyway, I guess I would suggest just using python to construct the TGeo configuration, which should be much more readable/maintainable than the JSON version in my opinon? (see [here](https://github.com/acts-project/acts/blob/main/Examples/Scripts/Python/itk.py#L112-L219))

File: Examples/Python/python/acts/_adapter.py (hunk 1 above)
Comment (paulgessinger) [reply Examples/Python/python/acts/_adapter.py:89]: Ah, this is because there's no getter, I see

Comment (timadye) [reply Examples/Python/python/acts/_adapter.py:89]: `hasttr()` tests `getattr()` works, not `setattr()`. In this case `jsonFile` declares the setter, not the getter, so wasn't picked up.

File: Examples/Python/src/Detector.cpp [hunk 2]
```diff
@@ -108,6 +108,10 @@ void addDetector(Context& ctx) {
 
     auto c = py::class_<Config>(d, "Config").def(py::init<>());
 
+    c.def_property(
+        "jsonFile", nullptr,
+        [](Config& cfg, const std::string& file) { cfg.readJson(file); });
+
```
Comment (paulgessinger) [inline Examples/Python/src/Detector.cpp:114]: Do I understand correctly that the reason for this is that you want to support passing `jsonFile` through the `create` interface?

**Comment by timadye** [comment]
> But if you're using the python bindings anyway, I guess I would suggest just using python to construct the TGeo configuration, which should be much more readable/maintainable than the JSON version in my opinon?

Agreed. I found this useful to load an existing ITk configuration in JSON format.

> (see [here](https://github.com/acts-project/acts/blob/main/Examples/Scripts/Python/itk.py#L112-L219))

That didn't work for me, so I added this option to compare. The JSON version works for me, so this helps as a fallback and/or to debug the Python one.

File: Examples/Python/src/Detector.cpp (hunk 2 above)
Comment (timadye) [reply Examples/Python/src/Detector.cpp:114]: Yes. This seemed to be the easiest way to implement it, and allows the other configs to still be set in Python with the same `create` interface.

**Review by paulgessinger — APPROVED** [review]

File: Examples/Python/python/acts/_adapter.py (hunk 1 above)
Comment (timadye) [reply Examples/Python/python/acts/_adapter.py:89]: Another way to work round this that I tried was to ctreate a dummy getter for `jsonFile`. I think this fix is neater as it solves the core problem.

**Pull request merged**